packages/superset-ui-preset-chart-xy/src/components/WithLegend.tsx

```diff
@@ -29,34 +11,32 @@ type Props = {
   legendJustifyContent: 'center' | 'flex-start' | 'flex-end';
   position: 'top' | 'left' | 'bottom' | 'right';
   renderChart: (dim: { width: number; height: number }) => ReactNode;
-  renderLegend: (params: { direction: string }) => ReactNode;
-  hideLegend: boolean;
+  renderLegend?: (params: { direction: string }) => ReactNode;
```

Comment:
I saw we have two different ways to define Props with defaultProps. Can we make the styles consistent? I do like the way you defined before, with `& typeof defaultProps`.

Reply:
Yes, fixed.